src/Todos.Service/TodoListsApi.cs

```diff
@@ -12,5 +12,49 @@ public static void MapTodosApi(this WebApplication app)
             return lists
                 .Select(todoList => new TodoListModel(todoList));
         });
+
+        app.MapPost("/todos", async ([FromServices] AppContext context, TodoListModel model) =>
+        {
+            var existingList = await context.TodoLists.FirstOrDefaultAsync(x => x.Name == model.Name);
+            if (existingList != null) return $"Lista o nazwie '{model.Name}' już istnieje.";
+
+            var newList = new ToDoList
+            {
+                Name = model.Name,
+                Description = model.Description
+            };
+            context.TodoLists.Add(newList);
+            await context.SaveChangesAsync();
+            return $"Utworzono nową listę o nazwie: {newList.Name}";
+        });
+
+        app.MapDelete("/todos/{name}", async ([FromServices] AppContext context, string name) =>
+        {
+            var listToDelete = await context.TodoLists.FirstOrDefaultAsync(t => t.Name == name);
+            if (listToDelete == null) return $"Lista o nazwie {name} nie istnieje";
+            else
+            {
+                context.TodoLists.Remove(listToDelete);
+                await context.SaveChangesAsync();
+                return $"Usunięto listę o nazwie: {name}";
+            }
+        });
+
+        app.MapPost("/todos/{list_name}/task", async ([FromServices] AppContext context, string list_name, ToDoListItemModel model) =>
+        {
+            var todoList = await context.TodoLists.FirstOrDefaultAsync(t => t.Name == list_name);
+            if (todoList == null) return $"Nie znaleziono listy o nazwie: {list_name}";
+            else
+            {
+                var newTask = new ToDoListItem
```

Comment:
Ujednoliciłbym nazewnictwo - skoro w adresie mamy słówko `task` to klasa powinna nazywać się `TodoListTask` (lub powinniśmy zmienić adres)